### ISSUE TITLE
Unexpected switch to a new tab

### ISSUE THREAD

**Issue by I-am-me** [mozilla/multi-account-containers#923]
When opening a new tab in Firefox you are not automatically redirected to it, you stay on the tab on which you were. But when opening a tab in a container, you are redirected to this tab.
For a more streamlined experience I think you should not be automatically redirected to the container when opening it.

Running Containers 4.0.3 on Firefox 56.0.2 on Windows 8.1

**Comment by cole-h** [comment]
When you say "when opening a new tab" do you mean from the M-A-C menu? I tried to replicate that in a fresh install of FF56 with Containers and when I click the container I want the new tab in, it instantly creates the new tab in the chosen container. Unless you mean you want this to replace the current tab's container with whatever new container tab you were trying to open.

**Comment by I-am-me** [comment]
My bad, I forgot to mention something really important.
You're right, when right-clicking and choosing "open in a new container tab" it behaves correctly, creating a new tab in the chosen container, and doesn't redirect to itself.
However if a website is assigned to "always open in this container" , when I right-click "open in a new tab" on a link to this website the newly created container tab redirects me to itself (which it should not do because I have disabled "when I open a link in a new tab, switch to it immediately").

**Comment by cole-h** [comment]
Oh, you were talking about the right click context menu, got it. By "redirects me to itself" do you also notice it opening 2 of the same tab? When trying to test out the "open in a new tab" on a tab I have set to always open in a container, I noticed the same thing: my test site (Twitter in this case) would open twice, but still in its correct container.

**Comment by I-am-me** [comment]
On my side it is not opening 2 of the same tab.
Just to clarify, what I mean by "redirects me to itself" is that once the container tab is created, Firefox switches me to it automatically.
I just did some tests and I discovered that if I choose "open in a new container tab" and choose the container associated with the website, the created tab will not redirect me to itself. But if I choose "open in a new tab", I will be redirected to the newly created container tab.
(Btw I'm testing with Facebook)

**Comment by biva** [comment]
I think what @I-am-me is saying, is that opening a link in a new tab should respect the parameter chosen in the FF options. For example, if I disabled "when I open a link in a new tab, switch to it immediately", I don't want that FF switches towards the new tab when I "middle-click" on a link.

Today, if I middle-click (or right click - context menu - open in a new context) on a link which is assigned to a specific container, it will open the link in this container, and SWITCH to it. It shouldn't switch.

@I-am-me , is that right? I think you may want to modify the title to something like "Unexpected switch to a new tab in some cases"

If not, I'll open a specific issue, because this bug is annoying me ;)

**Comment by I-am-me** [comment]
@biva This is exactly what I meant, thank you for expressing it in clearer terms that I could

**Comment by rgamici** [comment]
I also have noticed this.
It is respecting the option to open the new tabs and do not switch to them.
But if it has the rule to "Always open on...", it switches to that tab to confirm if it should open in the assigned container or in the one it was open (default or another one). 
I guess this is intentional (but an option to disable it would be welcome).

If you had previously selected the checkbox "Remember my decision for this site", when it assigns the selected container, it still changes the focus to that tab.

I can see a new tab opening in the background when I open it using ctrl+left click. When it resolves the url and it matches a container, it changes to it as it assigns the container (and the different color for that tab).
I guess it is still opening that confirmation screen (and changing the focus to it), but then bypass it immediately and assigns the container.

**Comment by AdrienPoupa** [comment]
+1 for this, a heavy use of different containers (eg one for Facebook, one for Github, one for Google) makes navigation a pain.

**Comment by groovecoder** [comment]
Just release 5.0.1 today which should have the fix for this.

**Comment by markiz** [comment]
@groovecoder thanks!
Is the issue mentioned by @rgamici a comment above yours fixable though?

**Comment by groovecoder** [comment]
I think so, yes ... but happy if someone can verify.

**Comment by markiz** [comment]
@groovecoder for me it still switches focus to new tab when opened from the bookmarks tab with middle click (although right-clicking and then using the context menu to open in a container works as I expect)

**Comment by rgamici** [comment]
@markiz I tried to reproduce the problems I have described earlier, and they seem to be fixed.

Does your bookmarks without a container open in the background?

Mine was also changing the focus, but it is for all bookmarks.
It is probably another issue, because when I checked the options in about:config, the parameter 'browser.tabs.loadBookmarksInBackground' had its original value (false), even with the option "When you open a link in a new tab, switch to it immediately" not set in the preferences.
As far as I remember (and from my previous post), it was opening the links from the bookmark toolbar in the background.
